cermod/r1face_mic/r1face_mic.cpp

```diff
@@ -88,94 +123,180 @@ bool R1faceMic::open(yarp::os::Searchable &params)
 
 bool R1faceMic::close()
 {
-    recording = false;
+    m_isRecording = false;
     stop();
     return true;
 }
 
 
 bool R1faceMic::threadInit()
 {
-    recording = true;
+    //better not to start the recording here!
+    //you may get buffer overrun if you do not call getSound() quickly enough
+    //startRecording();
     return true;
 }
 
 void R1faceMic::run()
 {
     // when not recording, do nothing
-    if(!recording)
+    if(!m_isRecording)
     {
         yarp::os::Time::delay(0.01);
         return;
     }
 
+#ifndef SIMULATE_ONLY
     // Just acquire raw data and place them in the buffer as fast as possible
     if (::read(dev_fd, tmpData->data, CHUNK_SIZE * HW_STEREO_CHANNELS * STEREO * sizeof(int32_t)) < 0)
     {
         yError() << "R1 face microphones: error reading data from HW";
         return;
     }
+#endif
+
+    for (size_t i = 0; i < TOT_SAMPLES; i++)
+    {
+        inputBuffer->write(tmpData->data[i]);
+    }
+
+//#define DEBUG_BUFFER_SIZE
+#ifdef  DEBUG_BUFFER_SIZE
+    yDebug() << inputBuffer->size().getBufferElements() << "/" << inputBuffer->getMaxSize().getBufferElements() << "Buffer Elements";
 
-    inputBuffer->write(*tmpData);
+    yDebug() << inputBuffer->size().getSamples()<< "/" << inputBuffer->getMaxSize().getSamples() << "Samples";
+#endif
 }
 
 void R1faceMic::threadRelease()
 {
     stopRecording();
 }
 
-bool R1faceMic::getSound(yarp::sig::Sound& sound)
+bool R1faceMic::getSound(yarp::sig::Sound& sound, size_t min_number_of_samples, size_t max_number_of_samples, double max_samples_timeout_s)
 {
-    ///////////////////////////////////////////////
-    // Extract channels from acquired buffer and
-    // manipulate them to get meaningful information
-
-    while(inputBuffer->size() < 5 && recording)
+    //check for something_to_record
     {
-        yarp::os::SystemClock::delaySystem(0.01);
+#ifdef AUTOMATIC_REC_START
```

Comment:
Personally I'd use as few `#ifdef` as possible. Can we choose one way of working and delete the other one?
When user remotely starts the device, for example from yarpmanager, he/she does not know how the device was compiled... I'd prefer to remove any source of uncertainty in device default behaviour.